packages/ethereum/contracts/src/node-stake/permissioned-module/CapabilityPermissions.sol

```diff
@@ -0,0 +1,760 @@
+        } else if (target.getTargetType() == TargetType.SEND) {
+            granularPermission = checkSendParameters(role, targetAddress);
+        }
+
+        // check permission result
+        if (
+            granularPermission == GranularPermission.BLOCK ||
+            (granularPermission == GranularPermission.NONE && defaultPermission == TargetPermission.SPECIFIC_FALLBACK_BLOCK)
+        ) {
+            revert GranularPermissionRejected();
+        } else if (
+            granularPermission == GranularPermission.ALLOW || 
+            (granularPermission == GranularPermission.NONE && defaultPermission == TargetPermission.SPECIFIC_FALLBACK_ALLOW)
+        ) {
+            return;
+        } else {
+            revert PermissionNotConfigured();
+        }
+    }
+
+    /**
+     * @dev Check if target is scoped; if the transaction can send along native tokens; if DelegatedCall is allowed.
+     * @param value The value of the transaction.
+     * @param operation The operation type of the transaction.
+     * @param target The stored target
+     */
+    function checkExecutionOptions(
+        uint256 value,
+        Enum.Operation operation,
+        Target target
+    ) internal pure {
+        if (target.getTargetClearance() != Clearance.FUNCTION) {
+            revert TargetAddressNotAllowed();
+        }
+
+         // delegate call is not allowed; 
+        if (
+            operation == Enum.Operation.DelegateCall
+        ) {
+            revert DelegateCallNotAllowed();
+        }
+        
+        // send native tokens is only available to a set of addresses
+        if (value > 0 && !target.isTargetType(TargetType.SEND)) {
+            revert SendNotAllowed();
+        }
+    }
+
+    /*
+     * @dev Check parameters for HoprChannels capability
+     * @param role reference to role storage
+     * @param capabilityKey Key to the capability.
+     * @param functionSig Function method ID
+     * @param data payload (with function signature)
+     */
+    function checkHoprChannelsParameters(
+        Role storage role,
+        bytes32 capabilityKey,
+        bytes4 functionSig,
+        bytes memory data
+    ) internal view returns (GranularPermission) {
+        // check the first two evm slots of data payload
+        // according to the following ABIs
+        //  - fundChannelSafe(address self, address account, Balance amount)  // src,dst
+        //  - redeemTicketSafe(address self, RedeemableTicket calldata redeemable) // dst,channelId
+        //  - initiateOutgoingChannelClosureSafe(address self, address destination) // src,dst
+        //  - closeIncomingChannelSafe(address self, address source) // dst,src
+        //  - finalizeOutgoingChannelClosureSafe(address self, address destination) // src,dst
+        //  - setCommitmentSafe(address self, address source, bytes32 newCommitment) // dst,src
```

Comment:
not necessary anymore

Reply:
If "set commitment" function will be entirely removed, it can be adapted, but only when your changes in Channels.sol are merged and reflects the change.